src/main/java/persistence/sql/dml/DMLQueryBuilder.java

```diff
@@ -1,29 +1,29 @@
 package persistence.sql.dml;
 
-import java.lang.reflect.Field;
+import persistence.entity.metadata.DefaultEntityMetadataReader;
+import persistence.entity.metadata.EntityColumn;
+import persistence.entity.metadata.EntityDataManipulator;
+
 import java.util.List;
 import java.util.stream.Collectors;
-import persistence.core.EntityContextManager;
-import persistence.entity.metadata.EntityColumn;
-import persistence.entity.metadata.EntityMetadata;
-import persistence.inspector.EntityInfoExtractor;
 
-public class DMLQueryBuilder extends EntityContextManager {
+public class DMLQueryBuilder {
 
-    private DMLQueryBuilder() {
-    }
+    private static DMLQueryBuilder dmlQueryBuilder = null;
 
     public static DMLQueryBuilder getInstance() {
-        return new DMLQueryBuilder();
+        if (dmlQueryBuilder == null) {
+            dmlQueryBuilder = new DMLQueryBuilder();
+        }
+        return dmlQueryBuilder;
     }
 
-    private final static String COLUMN_SEPARATOR = ", ";
+    private static final  String COLUMN_SEPARATOR = ", ";
 
     public String insertSql(Object entity) {
-        EntityMetadata entityMetadata = getEntityMetadata(entity.getClass());
-        String tableName = entityMetadata.getEntityTable().getTableName();
-        String columns = getColumnNamesClause(entityMetadata.getColumns().getInsertTargetColumns());
-        String columnValues = getColumnValueClause(entity, entityMetadata.getColumns().getInsertTargetColumns());
+        String tableName = DefaultEntityMetadataReader.getTableName(entity.getClass());
+        String columns = getColumnNamesClause(DefaultEntityMetadataReader.getInsertTargetColumns(entity.getClass()));
+        String columnValues = getColumnValueClause(entity, DefaultEntityMetadataReader.getInsertTargetColumns(entity.getClass()));
```

Comment:
QueryBilder 에 Object 를 던지기 보다 쿼리를 생성하는데 필요한 정보만 메시지로 전달하는건 어떻게 생각하세요?
저는 객체에 책임을 부여할 때 전달하는 메시지는 책임을 수행하는데 꼭 필요한 정보만 전달하는 것을 선호합니다.
객체 지향적으로나 함수지향적으로나 제한되고 정제된 인풋과 아웃풋은 책임을 더 명확하게 만들어 클린코드의 베이스가 된다고 생각하기 때문이에요.

```suggestion
    public String insertSql(EntityMeta entityMeta) {
        return dmlQueryFormatter.createInsertQuery(entityMeta.getTableName, entityMeta.getColumns, entityMeta.colunValues);
    };
```